eng/common/scripts/Save-Package-Properties.ps1

```diff
@@ -92,7 +92,18 @@ function GetRelativePath($path) {
 }
 
 $exportedPaths = @{}
-$allPackageProperties = Get-AllPkgProperties $serviceDirectory
+
+$allPackageProperties = @()
+
+if ($serviceInput.endswith("diff.json")) {
+  $allPackageProperties = Get-PrPkgProperties $serviceInput
+}
+else {
+  $allPackageProperties = Get-AllPkgProperties $serviceInput
```

Comment:
Would it be easier to get all the package infos (i.e. ignore the service directory) and then filter them outside of the common language settings?

Reply:
I could! I'll note that you explicitly asked me to use the Save-Package-Properties to filter.

This is what it would look like [to just save package properties for the affected services](https://github.com/azure/azure-sdk-for-python/compare/2ca0b6faf470130a92df6b31b431c31c10d4dfd0...06df40dbd4c2269368382ed79e16a26c436725dd), without filtering in Save-Package-Properties. It's a fairly straightforward operation to simply run Save-Package-Properties for each affected service, and then filter to the affected packages. [This is what the current PR looks like in practice](https://github.com/Azure/azure-sdk-for-python/compare/main...pipelinev3)

Reply:
I was thinking something like:

```
$allPackageProperties = Get-AllPkgProperties
$packagesWithChanges = @()

foreach($pkg in $allPackageProperties)
{
  foreach($file in $allFilesInDiff)
  {
    # would need to normalized the paths to verify it is under
    if ($file is in exists under $pkg.DirectoryPath) {
      $packagesWithChanges += $pkg
    }
  }
}
```

Then we write out all $packagesWithChanges

Reply:
This is extremely close what I already do frankly. Get-PRPackageProperties avoids going after the _entire_ repo worth of metadata. For python that saves ~30 seconds worth of parsing of all the package info.

Reply:
If that's all you've got an issue with, then I'll just do that and move on tho.

Reply:
My main goal is to ensure we only write out package info files for the packages we are running a testing as there will be a number of steps that depend on only having the exact set. Secondly, we do have the `Get-AllPkgProperties` implemented in all the languages already so we should be able to use that and implement this filtering in one place instead of having to implement it in each language independently.

Reply:
Alrighty. Did exactly this. [Working properly](https://dev.azure.com/azure-sdk/public/_build/results?buildId=3963068&view=logs&j=8e5196d1-9c2a-56bb-41fd-e329e85c48fa&t=6119a4c1-14d2-5fe9-3d3b-b95005807b68).

I'll deal with any fallout from `ServiceInput` vs `ServiceDirectory` and get this merged.